### ISSUE TITLE
Upload to parent project with no parent version

### ISSUE THREAD

**Issue by oallauddin** [pmckeown/dependency-track-maven-plugin#375]
My pom.xml has a parent pom and I don't want to use its name or version for the parent project in Dependency Track. The parentName I want to use is MY APP and the parentVersion I want to use is an empty string. When I use an empty string in the configuration maven treats this as a null and passes the default value from the mojo ${project.parent.version}. I don't want this because the project MY APP-2.00 does not exist in Dependency Track but the project MY APP with no version does exist in Dependency Track. Is there any way to get this working?

```
[ERROR] Failed to find parent project with name ['MY APP-2.00']. Check the update parent settings of your this plugin and verify if a matching parent project exists in the server.
```

Snippet of what the pom.xml looks like
```
   ...
    <parent>
        <groupId>org.build.maven</groupId>
        <artifactId>common-parent</artifactId>
        <version>2.00</version>
    </parent>
    ...
    <configuration>
        <updateProjectInfo>true</updateProjectInfo>
        <updateParent>true</updateParent>
        <parentName>MY APP</parentName>
        <parentVersion/>
    </configuration>
    ...
```

![parent_project](https://github.com/pmckeown/dependency-track-maven-plugin/assets/13156842/2b98ebd9-f0e7-4339-b53e-d3bbcf2b8b33)

**Comment by pmckeown** [comment]
Hi, the simple answer is to assign a version in Dependency Track to the parent.  That will get this working.

In the longer term, support for an empty version when setting the parent would need to be built in.

**Comment by roadSurfer** [comment]
I already had the changes in a branch that went stale, I've updated it and create a [PR](https://github.com/pmckeown/dependency-track-maven-plugin/pull/393) to (hopefully) address as I will also be impacted.

**Comment by pmckeown** [comment]
Fixed by #396
